rpcs3/Emu/Io/usio.cpp

```diff
@@ -434,50 +431,14 @@ void usb_device_usio::usio_read(u8 channel, u16 reg, u16 size)
 	}
 	else if (channel >= 2)
 	{
-		u8 chip = channel - 2;
+		const u8 chip = channel - 2;
 		usio_log.trace("Usio read of sram(chip: %d, addr: 0x%04X)", chip, reg);
-		switch (chip)
-		{
-		case 0:
-		{
-			switch (reg)
-			{
-			case 0x0000:
-			{
-				response = g_fxo->get<usio_memory>().backup_memory;
-				break;
-			}
-			case 0x0180:
-			{
-				response = g_fxo->get<usio_memory>().last_game_status;
-				break;
-			}
-			case 0x0200:
-			{
-				//ensure(size == 0x100);
-				// No data returned
-				break;
-			}
-			case 0x1000:
-			{
-				//ensure(size == 0x1000);
-				// No data returned
-				break;
-			}
-			default:
-			{
-				usio_log.error("Unhandled read of sram(chip: %d, addr: 0x%04X)", channel - 2, reg);
-				break;
-			}
-			}
-			break;
-		}
-		default:
-		{
-			usio_log.error("Unhandled read of sram(chip: %d, addr: 0x%04X)", channel - 2, reg);
-			break;
-		}
-		}
+		auto& memory = g_fxo->get<usio_memory>().backup_memory;
+		const usz addr_end = reg + size;
+		if (size > 0 && chip < usio_memory::chip_count && addr_end < usio_memory::chip_size)
+			response.insert(response.end(), &memory[usio_memory::chip_size * chip + reg], &memory[usio_memory::chip_size * chip + addr_end]);
```

Comment:
```suggestion
			response.insert(response.end(), memory.begin() + (usio_memory::chip_size * chip + reg), memory.begin() + (usio_memory::chip_size * chip + addr_end));
```
Use of iterators to avoid UB when referring to last element. (instead of &memory[chip_size * chip])

Reply:
I see. Thank you for pointing it out.